webroot/src/pages/FinalizePrivilegePurchase/FinalizePrivilegePurchase.ts

```diff
@@ -139,11 +139,20 @@ export default class FinalizePrivilegePurchase extends mixins(MixinForm) {
     get stateOptions() {
         const stateOptions = [{ value: '', name: this.$t('common.select') }];
 
-        const states = this.$tm('common.states') as Array<any>;
+        let states = this.$tm('common.states') as Array<any>;
+
+        if (typeof states[0]?.abbrev === 'function') {
+            const normalize = ([value]) => value;
+
+            states = states.map((st) => ({
+                abbrev: st.abbrev({ normalize }),
+                full: st.full({ normalize }),
+            }));
+        }
```

Comment:
Oh got it, that's better than what I did!

Reply:
So this is another example of the ts never bug: 
<img width="2560" alt="Screenshot 2024-11-20 at 5 00 11 PM (2)" src="https://github.com/user-attachments/assets/7294afcb-217d-4a7f-88a1-13d405764415">

Reply:
This single ts bug has been a thorn in my side many times, I would like to maybe add add a global ignore or rule, I have refactored a fair amount to avoid it. How do you feel about that?

Reply:
I guess we do this: https://stackoverflow.com/questions/75677237/silence-typescript-error-property-does-not-exist-on-type-never

Reply:
Actually, back to the actual code block we are talking about: that code doesn't seem to work because before normalization the abbrev is not accessible so we cant do: `const model = new State({ abbrev: state.abbrev });`